Tests/ZIPFoundationTests/ZIPFoundationReadingTests.swift

```diff
@@ -269,4 +269,22 @@ extension ZIPFoundationTests {
         let archive = self.archive(for: #function, mode: .update)
         XCTAssert(archive.totalUnitCountForAddingItem(at: nonExistantURL) == -1)
     }
+
+    func testExtractUncompressedEmptyFile() {
+        let expectation = XCTestExpectation(description: "Called extract consumer")
+        let archive = self.archive(for: #function, mode: .read)
+        guard let entry = archive["empty.txt"] else { XCTFail("Failed to extract entry."); return }
+        let progress = archive.makeProgressForReading(entry)
+        do {
+            _ = try archive.extract(entry, progress: progress) { (data) in
+                // Progress does not finished because Progress assumes totalUnitCount > 0
+                // XCTAssertFalse(progress.isFinished)
```

Comment:
When extract 0byte entry, `Progress#isFinished` returns false.
I don't know it is good way or not...
https://developer.apple.com/documentation/foundation/progress

IMO, It is intuitive that Progress has 1/1 progress for extracting 0byte entry.

Reply:
This is a good point - we currently handle this edge case properly. 
We are using a 1/1 progress for entries of type `.directory` already and could also use the same approach for zero-byte files.  

Since this PR handles the bug where the completion handler is not called for zero-byte files, I removed the progress related code from the test case you provided. I think the progress mis-reporting should be handled in a dedicated PR.